sqlglot/helper.py

```diff
@@ -53,7 +117,18 @@ def subclasses(module_name, classes, exclude=()):
     ]
 
 
-def apply_index_offset(expressions, offset):
+def apply_index_offset(expressions: t.List[E], offset: int) -> t.List[E]:
```

Comment:
why E? why not just use Expression you have it in t.TYpeChecking now

Reply:
I recommended this.
```python
l = [exp.Table(...), exp.Table(...)  # this is list[exp.Table]
apply_index_offset(l)  # using TypeVar, the return type is exp.Table. Without it, it's exp.Expression
```

I know, I know. You love type annotations Toby.

Reply:
ah i see